flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -129,6 +129,16 @@ public boolean isJobManagerPortReady(Configuration config) {
         return false;
     }
 
+    public boolean isJobManagerServing(Configuration config) {
+        try (ClusterClient<String> clusterClient = getClusterClient(config)) {
+            clusterClient.listJobs().get(10, TimeUnit.SECONDS);
+            return true;
```

Comment:
Does this have another way to check the JobManager is serving? The exception way doesn't look good to me.

Reply:
I do agree the try catch solution is not elegant but this pr aims at verifying the REST service of JM.
For network issue, to my best knowledge, try catch is the most widely used choice. Do you have any suggestion?

Reply:
@bgeng777, what about using the `clusterClient#getWebInterfaceURL` to check the serving?

Reply:
`getWebInterfaceURL` is neat but one drawback of `clusterClient#getWebInterfaceURL` is that it only returns the current web url(i.e. address of the leader of jm) but does not guarantee the JM can actually work. In `isJobManagerPortReady`, we currently have already used `getWebInterfaceURL` and will 'telnet' the port in the url. In the jira, we think it may not be enough and that's why I finally choose the `listJob` call.

Reply:
The `getWebInterfaceURL` might not work in the current situation because we are using `StandaloneClientHAServices` when creating the rest cluster client.

Reply:
Please move the timeout into the config, the previous delay was also configurable. Did you check how the timeout behaves? If REST API is not available when the call is made and then becomes available after 5s, will the list call still fail after waiting for 10s? I believe that's what I had seen when testing this.

Reply:
hi @tweise, I check the flink code and find the underlying impl of `listJob` is [retriable](https://github.com/apache/flink/blob/46bb03848ea67fa7b8952f757d57a2cda6ab16aa/flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java#L833). The retry delay is controlled by the flink config `rest.retry.delay` (default val is 3s). For your case, if the initial call falis, it should retry each 3s until achieving the max retry attempts(default is 20). So do you mean you meet unexpected failure of this list call? Any steps that I can follow to repeat it? 

For the configurable timeout, in fact, I reuse the `listJobs()` in our `FlinkService`. I want to know if you are meaning that we should make both of them wait a configurable timeout. thanks.

Reply:
Hi @tweise , I tried following steps to test the actual behavior of `clusterClient.listJobs()`:
1. Apply the `basic-session-example` yaml and wait for everything to be fine
2. Run ` kubectl scale deployment basic-session-example --replicas=0` so JM port and rest call are both unavailable
3. Rescale the replicas to 1 quickly when the `isJobManagerServing` is waiting.
I keep monitoring the operator log (I added some local dummy debug info)
<img width="1339" alt="image" src="https://user-images.githubusercontent.com/80749729/158746332-dff95fec-2701-4b35-b39e-3eb60d647c64.png">

From the log, I find that the rest call is sent at 13:56:05(In the k8s log, it is 05:56:05 due to time lag) and keeps waiting. Then when JM is started at 13:56:07 and everything becomes ok again around 13:56:13, the rest call succeeds. So I believe the answer to "will the list call still fail after waiting for 10s" is that it will not fail due to the retry.